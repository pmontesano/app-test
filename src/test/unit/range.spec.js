/**
 * @jest-environment jsdom
 */

import React from 'react';
import { create } from 'react-test-renderer';
import { mount, shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Range from '../../components/range';

configure({ adapter: new Adapter() });

const props = {
  min: 1,
  max: 100,
  onChange: jest.fn(),
};

describe('Range', () => {
  let wrapper;
  let container;
  let requestAnimationFrame;

  beforeEach(() => {
    wrapper = shallow(<Range {...props} />);

    requestAnimationFrame = window.requestAnimationFrame;
    window.requestAnimationFrame = (callback) => callback();

    container = document.createElement('div');
    document.body.appendChild(container);
  });

  it('should render correctly', () => {
    const wrapper = create(<Range {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('initial render values min and max must be equal 1 and 100', () => {
    const wrapper = shallow(<Range {...props} />);

    const minValue = wrapper.find('input').at(0);
    const maxValue = wrapper.find('input').at(1);

    expect(minValue.props().value).toEqual(1);
    expect(maxValue.props().value).toEqual(100);
  });

  describe('changing minValue', () => {
    beforeEach(() => {
      wrapper = shallow(<Range {...props} />);
      const minValue = wrapper.find('input').at(0);

      minValue.simulate('change', {
        target: { value: 50 },
      });
    });

    it('should be change minValue', () => {
      const minValue = wrapper.find('input').at(0);
      expect(minValue.props().value).toEqual(50);
    });
  });

  describe('changing maxValue', () => {
    beforeEach(() => {
      wrapper = shallow(<Range {...props} />);
      const maxValue = wrapper.find('input').at(0);

      maxValue.simulate('change', {
        target: { value: 80 },
      });
    });

    it('should be change maxValue', () => {
      const maxValue = wrapper.find('input').at(0);
      expect(maxValue.props().value).toEqual(80);
    });
  });
});
