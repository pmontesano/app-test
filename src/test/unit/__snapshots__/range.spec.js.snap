// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range should render correctly 1`] = `
<div
  className="range-slider-wrapper"
>
  <label
    className="range-slider-label range-slider-label--left"
    htmlFor="rangeValueMin"
  >
    <input
      className="range-slider-input"
      id="rangeValueMin"
      onChange={[Function]}
      type="number"
      value={1}
    />
    €
  </label>
  <div
    className="range-slider"
  >
    <div
      className="range-slider__thumb"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      <button
        className="range-slider__handle range-slider__handle--left "
        style={
          Object {
            "zIndex": false,
          }
        }
        value={1}
      />
    </div>
    <div
      className="range-slider__thumb"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      <button
        className="range-slider__handle range-slider__handle--right "
        style={
          Object {
            "marginLeft": "-10px",
          }
        }
        value={100}
      />
    </div>
    <div
      className="range-slider__track"
    />
  </div>
  <label
    className="range-slider-label range-slider-label--right"
    htmlFor="rangeValueMax"
  >
    <input
      className="range-slider-input"
      id="rangeValueMax"
      onChange={[Function]}
      type="number"
      value={100}
    />
    €
  </label>
</div>
`;
